### ISSUE TITLE
1.3.x cherries 🍒 

### ISSUE THREAD

**Issue by carmocca** [PyTorchLightning/pytorch-lightning#7467]
<details>
<summary>1.3.1 (#7477)</summary>

https://github.com/PyTorchLightning/pytorch-lightning/pull/7456 6d82dc832b066a9d6a5fa63a449b8ac372d72b6a Pin `Sphinx<4.0`
https://github.com/PyTorchLightning/pytorch-lightning/pull/7395 6bc616d78f13c9921f3a08f7c71229b81be8b5ca fix local rank display bug in info message
https://github.com/PyTorchLightning/pytorch-lightning/pull/7441 1af42d7d1e3f02d78063e02b5a603e3dd59837eb Update worker_init_fn test for pytorch 1.9 nightly
https://github.com/PyTorchLightning/pytorch-lightning/pull/7434 710b144b9ba7d5e1cbc4bc3817ba2d30d6e8968d Restore `trainer.current_epoch` after tuning
https://github.com/PyTorchLightning/pytorch-lightning/pull/7414 1a27c12b26ef1fc94ddd2cfed6b00238c73af88a update ngc for 1.3
https://github.com/PyTorchLightning/pytorch-lightning/pull/7362 98b94b810c89d5e51a0ad0a2e6a87747aee6fbe9 Fix DeepSpeedPlugin with IterableDataset

</details>

<details>
<summary>1.3.2 (#7589)</summary>

#7032 20f63377f81f4771d3f128f979b3a0f9b8d219a7 Fix the condition for calling update_learning_rates
https://github.com/PyTorchLightning/pytorch-lightning/pull/7492 6ac16ff34822cef9b3c16e54f872655b585a066a Fix DistribType for `ddp_cpu` (spawn)
https://github.com/PyTorchLightning/pytorch-lightning/pull/7433 7b283e3c46c014affe1de1f6892bf310aebd886a Bugfix/Multiple dataloaders
#7238 b65ae7947803e7dc2a254af1045c1bcc6e4cf147 Automatically check `DataModule.has_{setup,teardown,prepare_data}` [2/2] 

</details>

<details>
<summary>1.3.3 (#7712)</summary>

#7692 8ba6304c73c0fd030c50d5de52cbaddfed161613 Increment the total batch idx before the accumulation early exit
#7683 2c10ecc232f5fb576638683e24744a5be8911c9a MAINTAINER has been deprecated
#7677 8b01497e4235139edb83cd0b8efd87380a1af726 Fix global step update when the epoch is skipped
#7415 3f460b150aab16d1fa267251d738bada6d890675 Move parameter validation specific to TPU Training plugins
#7674 a54bc5dba3133cf492205e6a60ca46e8ab22c465 Fix progress bar print error when called before training
#7566 0c958c5a1f8ca8926eeaf1a2035725c15417830e Fix dataloaders are not reset when tuning the model
#7563 01109cdf0c44a150c262b65e70a7e1e64003cf93 Fix/mismatched toggle optimizer
#7592 92cf396de2fe49e89a625a200d641bd8b6aeb328 Override `broadcast_object_list` for `torch<1.8`
#7608 ed271905cf07d86af5b6d8e088b6c04813d1a8a1 Clear predict_progress_bar in `ProgressBar.__getstate__`

</details>

<details>
<summary>1.3.4 (#7786)</summary>

#7781 0b6fd1da5458a0ef56f898b03e5caeb63489e9eb Update pre-commit and add new hooks
#7780 7e6010fc936f2e9196ac497e8b3f1c00957b5a96 fix info message when max training time reached
#7681 04dcb1786d0ea41dfdb1e73493b214169e365c6c Add `__len__` method to IndexBatchSamplerWrapper

</details>

<details>
<summary>1.3.5 (#7881)</summary>

https://github.com/PyTorchLightning/pytorch-lightning/pull/7868 f9fccdfb394bc35e33182996c5b11b00a86c4248 Move `training_output` validation to after `train_step_end`
https://github.com/PyTorchLightning/pytorch-lightning/pull/7869 ce976769ef2241230f13f0964a5f38dc49fdc923 update fsspec to 2021.06.0
https://github.com/PyTorchLightning/pytorch-lightning/pull/7835 d1becce4c16aeb391526c3a1e2caae6f4efd0332 [bugfix] Resolve LearningRateMonitor + BackboneFinetuning
https://github.com/PyTorchLightning/pytorch-lightning/pull/7851 009e05d14f9f8c63f401ccda565700dfa78a24a2 [bugfix] Minor improvements to `apply_to_collection` and type signature of `log_dict`
https://github.com/PyTorchLightning/pytorch-lightning/pull/7834 7b531ac7ac86141237f758a7e7446930964b91a9 fix docker versions
https://github.com/PyTorchLightning/pytorch-lightning/pull/7779 6a0d50369382ec3e7cce681facc9a2aad2a71d00 Add warning to trainstep output
https://github.com/PyTorchLightning/pytorch-lightning/pull/7825 ca89a7f3443710a2313e97b91ac908867eb33be5 [sharded plugin] Fix check for fp16 precision
https://github.com/PyTorchLightning/pytorch-lightning/pull/7807 f34584001c7a88aa7204ade4f0145ba109e5f1bc Fix support for torch Module type hints in LightningCLI

</details>

<details open=true>
<summary>1.3.6 (#7986)</summary>

https://github.com/PyTorchLightning/pytorch-lightning/pull/7993 92024df Pt 1.9 breaking fix: __iter__ type hint
https://github.com/PyTorchLightning/pytorch-lightning/pull/7931 3a0ed02bd46dc9f1fa39f98fde635a5d44b1b609 Properly handle parent modules w/ parameters in `BaseFinetuning` callback
https://github.com/PyTorchLightning/pytorch-lightning/pull/7958 cca0e7535ab9e67a5f9913116acc23c64d3a49b8 remove parsing comments
https://github.com/PyTorchLightning/pytorch-lightning/pull/7942 22d826615f525f45510905a7105f6a2f40b1dc47 Seed all workers when using DDP
https://github.com/PyTorchLightning/pytorch-lightning/pull/7967 20a5e09e33dc1a28fb74c3722e84e130ff85e7d7 fix myst-parser warning blocking docs ci
https://github.com/PyTorchLightning/pytorch-lightning/pull/7945 6856ccedfdab009b28d8b507068afdd00ac924fd Remove rank_zero_only on DataModule prepare_data
https://github.com/PyTorchLightning/pytorch-lightning/pull/7889 7f4ef6d135b8b824d23be9fe2ab21f184b12ea9b Fix logs overwriting issue for remote fs

`gh pr list -s merged -S 'merged:2021-06-08T16:30:00.000Z..2021-06-15T12:00:00.000Z' --json mergedAt,milestone,url,mergeCommit,title --jq 'sort_by(.mergedAt) | reverse | .[] | select((.milestone.title == "v1.3.x") or (.milestone.title == null)) | [.url, .mergeCommit.oid, .title] | join(" ")' --limit 100`

</details>

Including non-bugfix commits is allowed if they are necessary to support a bugfix commit and it does not break compatibility. In that case, please add a comment on why the commit was included.

---

### Utils

**Update the merged timestamp range as appropriate**

Requires gh version >=1.10.2

#### Commits

The following command provides a list of commits to master (ordered by merge date from newest to oldest) with the appropriate milestone (null or v1.3.x).

```console
gh pr list -s merged -S 'merged:2021-06-01T16:30:00.000Z..2021-06-08T16:30:00.000Z' --json mergedAt,milestone,url,mergeCommit,title --jq 'sort_by(.mergedAt) | reverse | .[] | select((.milestone.title == "v1.3.x") or (.milestone.title == null)) | [.url, .mergeCommit.oid, .title] | join(" ")' --limit 100
```

#### Author list

For a bug-fix release

```console
gh pr list -s merged -S 'merged:2021-06-08T16:30:00.000Z..2021-06-15T12:00:00.000Z' --json author,milestone --jq '.[] | select((.milestone.title == "v1.3.x") or (.milestone.title == null)) | ["@" + .author.login] | join("")' --limit 100 | sort --ignore-case | uniq | xargs echo 
```

For a feature release

```console
gh pr list -s merged --json author,milestone --jq '.[] | select(.milestone.title == "v1.4") | ["@" + .author.login] | join("")' --limit 8000 | sort --ignore-case | uniq | xargs echo
```

**Comment by Borda** [comment]
@carmocca the script is really GREAT!!

**Comment by carmocca** [comment]
Thank you @Borda 😊

**Comment by tchaton** [comment]
@carmocca I second that ! Really neat !
